cdap-support-bundle/src/main/java/io/cdap/cdap/support/services/SupportBundleGenerator.java

```diff
@@ -218,4 +308,50 @@ private NamespaceId validNamespace(NamespaceId namespace) throws Exception {
 
     return namespace;
   }
+
+  private SupportBundlePipelineStatus collectSupportBundleTaskStatus(
+    Set<SupportBundleTaskStatus> supportBundleTaskStatusSet) {
+    CollectionState systemLogTaskStatus = CollectionState.INVALID;
+    CollectionState pipelineInfoTaskStatus = CollectionState.INVALID;
+    CollectionState runtimeInfoTaskStatus = CollectionState.INVALID;
+    CollectionState runtimeLogTaskStatus = CollectionState.INVALID;
+    CollectionState vmInfoTaskStatus = CollectionState.INVALID;
+    // Expand all supportBundleTaskStatusSet and map each individual taskType and subTaskType with corresponding status
```

Comment:
nit: Please add a comment stating that the SupportBundleRuntimeInfoTask and SupportBundleRuntimeLogTask are two sub tasks of SupportBundlePipelineInfoTask.